crates/natrix/tests/actual_tests/derive_macros.rs

```diff
@@ -0,0 +1,221 @@
+use natrix::prelude::*;
+use wasm_bindgen_test::{wasm_bindgen_test, wasm_bindgen_test_configure};
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+// Test the Project derive macro with additional derives
+#[derive(Project)]
+#[project(Debug)]
```

Comment:
Not totally sold on this syntax for adding derives.

honestly the whole generinng a companion type is annoying, but its ofc the only good way of actually making this deriavable for most useful user enums (very few application enums only contain generic types.)

anyway, I think a better option is making `#[project(...)]` add those attrs to the generated enum, so `#[project(derive(...))]`, `#[project(serde(...))]` etc.